python/cudf/cudf/core/index.py

```diff
@@ -313,9 +313,20 @@ def copy(self, name=None, deep=False, dtype=None, names=None):
             Ignored for RangeIndex
         dtype : numpy dtype optional (default: None)
             Target dtype for underlying range data
+
+            .. deprecated:: 23.02
```

Comment:
We've never used this tag before, have we? OTOH, we do occasionally use the `Warns` section to show that we're going to raise a FutureWarning. I'm not opposed do doing this, jw if we should have a consistent approach (indicate warning, use the deprecated tag here, or both).

Reply:
I think it's best to do both. I myself got to know about [admonitions](https://pydata-sphinx-theme.readthedocs.io/en/stable/examples/kitchen-sink/admonitions.html) very recently. They get rendered nicely too. So felt it would be beneficial to have the deprecated admonition show up in docs too.

Reply:
Yep admonitions in Sphinx are nice. Happy to use both if you think both is best. Currently I don't see warnings documented, though; did you actually intend to include both, or only have the admonition?

Reply:
In this specific case `dtype` was already deprecated.